### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by mohitsoni**
1. Updated YAML Quick Start guide
2. Added new Java Quick Start Guide (#336)
3. Added Glossary

### CODE REVIEW

File: docs/tutorials/quick-start-java.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+And, finally change the directory to `/dcos-commons`, this is where the source code from your host is mounted inside the Vagrant box:
+
+```bash
+$ cd /dcos-commons/
+(dcos-commons)$ 
+```
+
+### Step 2 - Initialize service project
+Get started by forking: https://github.com/mesosphere/dcos-commons, and cloning it on your workstation. Change your working directory to `dcos-commons` and then issue following command to create a new project:
```
Comment (sascala) [inline docs/tutorials/quick-start-java.md]: Didn't we already do this in step 0?

File: docs/glossary.md [hunk 2]
```diff
@@ -42,6 +42,11 @@ Strategy determines whether children of a given Plan (i.e., Phase) or Phase (i.e
 ### Task
 A Task is a unit of execution of work.
 
+## U
+
+### Unit of Deployment
```
Comment (mohitsoni) [inline docs/glossary.md]: I think `Unit of deployment` is just a Pod. Also, `Unit of deployment` is not a first class terminology in SDK docs as yet. So, maybe we don't need to include it here.

**Comment by sascala** [comment]
@mohitsoni if the rest looks good to you, I'll merge

**Comment by mohitsoni** [comment]
@sascala LGTM!

**Comment by sascala** [comment]
retest this please

**Comment by keithchambers** [comment]
Based on user feedback, we need to improve the README Quick Start to be less "magic".

Before we merge this -- let's discuss how we want to improve the README Quick Start and then we can create a Java specific Quick Start.

**Comment by sascala** [comment]
retest this please

**Comment by sascala** [comment]
OK, @keithchambers. I can look it over for places to reduce "magic".

**Pull request merged**